api/src/opentrons/protocol_api/instrument_context.py

```diff
@@ -174,28 +175,36 @@ def aspirate(
         )
 
         well: Optional[labware.Well]
-        last_location = self._protocol_core.get_last_location()
+        move_to_location: types.Location
+        aspirate_in_place: bool = False
 
-        if isinstance(location, labware.Well):
-            move_to_location = location.bottom(z=self._well_bottom_clearances.aspirate)
-            well = location
-        elif isinstance(location, types.Location):
-            move_to_location = location
-            _, well = move_to_location.labware.get_parent_labware_and_well()
-        elif location is not None:
-            raise TypeError(
-                "location should be a Well or Location, but it is {}".format(location)
+        last_location = self._get_last_location_by_api_version()
+        try:
+            target = validation.validate_location(
+                location=location, last_location=last_location
             )
-        elif last_location:
-            move_to_location = last_location
-            _, well = move_to_location.labware.get_parent_labware_and_well()
-        else:
+        except validation.NoLocationError:
             raise RuntimeError(
                 "If aspirate is called without an explicit location, another"
                 " method that moves to a location (such as move_to or "
                 "dispense) must previously have been called so the robot "
                 "knows where it is."
             )
+        except validation.LocationTypeError:
+            raise TypeError(
+                "location should be a Well or Location, but it is {}".format(location)
+            )
```

Comment:
Raise always remove `"".format` if you're touching code

```suggestion
        except validation.LocationTypeError as e:
            raise TypeError(
                f"location should be a Well or Location, but it is {location}"
            ) from e
```